audio/audio_hw.c

```diff
@@ -1254,10 +1331,31 @@ static int in_set_parameters(struct audio_stream *stream, const char *kvpairs)
     return 0;
 }
 
-static char * in_get_parameters(const struct audio_stream *stream,
+static char* in_get_parameters(const struct audio_stream *stream,
                                 const char *keys)
 {
-    return strdup("");
+    ALOGV("+in_get_parameters(%p) '%s'", stream, keys);
+
+    struct stream_in_common *in = (struct stream_in_common *)stream;
+    struct str_parms *query = str_parms_create_str(keys);
+    struct str_parms *reply = str_parms_create();
+    char value[BUFFER_SIZE];
+    char *str;
+
+    if (str_parms_get_str(query, AUDIO_PARAMETER_STREAM_SUP_FORMATS,
```

Comment:
Lines 1345-1349 are identical to 465-469, apart from the value out->format or in->format.
This could all move into str_parms_add_audio_format() to remove the duplicated code.

Reply:
Done.